.github/workflows/build-and-deploy.yml

```
@@ -52,22 +52,25 @@ jobs:
-          asset_content_type: application/zip
+      - name: Install GitHub CLI
+        run: sudo apt-get install gh
+
+      - name: Upload war file to GitHub release
+        run: gh release upload ${{ github.event.release.tag_name }} target/*.war --clobber
+        env:
+          GITHUB_TOKEN: ${{ secrets.RELEASE_TOKEN }}
 
   create-release:
-    needs: build # This job depends on the build job
+    needs: [ build, upload-release-asset ] # This job depends on the build job
```

yamllint

[error] too many spaces inside brackets

(brackets)